vertx-web/src/main/java/io/vertx/ext/web/impl/RouteImpl.java

```diff
@@ -463,4 +469,16 @@ synchronized protected boolean hasNextContextHandler(RoutingContextImplBase cont
   synchronized protected boolean hasNextFailureHandler(RoutingContextImplBase context) {
     return context.currentRouteNextFailureHandlerIndex() < failureHandlers.size();
   }
+
+  private void handleDecodingError(RoutingContextImplBase context, Throwable e) {
+    if (router.getDecoderErrorHandler() == null)
+      context.response()
+        .putHeader(HttpHeaders.CONTENT_TYPE, "text/plain")
+        .setStatusMessage("Bad Request")
```

Comment:
can't we set code and or request using netty HttpResponseStatus.BAD_REQUEST - it carries both the code as int, and the reason phrase which is the status message.
does content-type has any value for 400 response ?